client/my-sites/sidebar-unified/collapse-sidebar.jsx

```diff
@@ -16,21 +16,23 @@ import PropTypes from 'prop-types';
 import SidebarItem from 'calypso/layout/sidebar/item';
 import SidebarCustomIcon from 'calypso/layout/sidebar/custom-icon';
 import { getSidebarIsCollapsed } from 'calypso/state/ui/selectors';
-import { collapseSidebar, expandSidebar } from 'calypso/state/ui/actions';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { savePreference } from 'calypso/state/preferences/actions';
 import TranslatableString from 'calypso/components/translatable/proptype';
 
 export const CollapseSidebar = ( { title, icon } ) => {
 	const reduxDispatch = useDispatch();
 	const sidebarIsCollapsed = useSelector( getSidebarIsCollapsed );
 
+	const onNavigate = () => {
+		reduxDispatch( recordTracksEvent( 'calypso_toggle_sidebar' ) );
+		reduxDispatch( savePreference( 'sidebarCollapsed', ! sidebarIsCollapsed ) );
+	};
+
 	return (
 		<SidebarItem
 			className="collapse-sidebar__toggle"
-			onNavigate={
-				sidebarIsCollapsed
-					? () => reduxDispatch( expandSidebar() )
-					: () => reduxDispatch( collapseSidebar() )
-			}
+			onNavigate={ () => onNavigate() }
```

Comment:
We can remove the additional callback here

```suggestion
			onNavigate={ onNavigate }
```

Reply:
Done e524308